### PR TITLE
Configure Azure Pipelines (YAML)

### PR DESCRIPTION
**Pull request by cafour**
Adds and modifies scripts, configs, csprojs, and other files so that our C# unit tests, Jest tests, and UI tests can run in Azure Pipelines.

(This is the complete implementation and thus closes #1046.)

### CODE REVIEW

File: ci/linux/test.ps1 [hunk 1]
```diff
@@ -0,0 +1 @@
+docker run --rm -it --mount "type=bind,src=${PWD},dst=/s" docker-reg.riganti.cz/docker-agent:latest-linux bash
```
Comment (MichalTichy) [inline ci/linux/test.ps1:1]: This accesses internal docker registry.

File: ci/linux/test.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+docker run --rm -it --mount "type=bind,src=${PWD},dst=/s" docker-reg.riganti.cz/docker-agent:latest-linux bash
```
Comment (MichalTichy) [inline ci/linux/test.sh:1]: This accesses internal docker registry.

Comment (MichalTichy) [reply ci/linux/test.sh:1]: Not sure if we event need this script.

File: ci/linux/test.ps1 (hunk 1 above)
Comment (MichalTichy) [reply ci/linux/test.ps1:1]: Not sure if we event need this script.

File: ci/windows/test.ps1 [hunk 3]
```diff
@@ -0,0 +1 @@
+docker run --rm -it --mount "type=bind,src=${PWD},dst=C:\s" docker-reg.riganti.cz/docker-agent:latest-win powershell
```
Comment (MichalTichy) [inline ci/windows/test.ps1:1]: This accesses internal docker registry.

File: src/DotVVM.Framework/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,6429 @@
+{
```
Comment (MichalTichy) [inline src/DotVVM.Framework/package-lock.json:1]: I dont think that this file should be part of repository

File: src/DotVVM.Samples.Tests/Feature/ApiTests.cs [hunk 5]
```diff
@@ -202,7 +202,7 @@ public void Feature_Api_AzureFunctionsApiTable()
             });
         }
 
-        [Fact]
+        [Fact(Skip = "Doesn't work on OWIN because it relies on _apiCore.")]
```
Comment (MichalTichy) [inline src/DotVVM.Samples.Tests/Feature/ApiTests.cs:205]: In the future it would be nice to have SkipOnOwin / SkipOnAspNetCore attributes.

**Review by MichalTichy — CHANGES_REQUESTED** [review]

File: src/DotVVM.Framework/package-lock.json (hunk 4 above)
Comment (cafour) [reply src/DotVVM.Framework/package-lock.json:1]: Why? It guarantees that everyone uses the same dependencies, which seems especially desirable in a CI setting.
(See https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json.)

File: src/DotVVM.Samples.Tests/Feature/ApiTests.cs (hunk 5 above)
Comment (cafour) [reply src/DotVVM.Samples.Tests/Feature/ApiTests.cs:205]: Agreed. Please create a separate issue.

**Review by quigamdev — APPROVED** [review]

**Pull request merged**